modules/dnn/perf/perf_layer.cpp

```diff
@@ -55,7 +55,197 @@ struct Layer_Slice : public TestBaseWithParam<tuple<Backend, Target> >
     }
 };
 
+struct Layer_NaryEltwise : public TestBaseWithParam<tuple<Backend, Target> >
+{
+    void test_layer(const std::vector<int>& a_shape, const std::vector<int>& b_shape, const String op)
+    {
+//        int backendId = get<0>(GetParam());
+//        int targetId = get<1>(GetParam());
+
+        Mat a(a_shape, CV_32FC1);
+        Mat b(b_shape, CV_32FC1);
+
+        Scalar mean = 0.f;
+        Scalar std = 1.f;
+        randn(a, mean, std);
+        randn(b, mean, std);
+
+
+        Net net;
+        LayerParams lp;
+        lp.type = "NaryEltwise";
+        lp.name = "testLayer";
+        lp.set("operation", op);
+        int id = net.addLayerToPrev(lp.name, lp.type, lp);
+        net.connect(0, 1, id, 1);
+
+        // warmup
+        {
+            std::vector<String> inpNames(2);
+            inpNames[0] = "a";
+            inpNames[1] = "b";
+            net.setInputsNames(inpNames);
+            net.setInput(a, inpNames[0]);
+            net.setInput(b, inpNames[1]);
+
+//            net.setPreferableBackend(backendId);
+//            net.setPreferableTarget(targetId);
+            Mat out = net.forward();
+        }
+
+        TEST_CYCLE()
+        {
+            Mat res = net.forward();
+        }
+
+        SANITY_CHECK_NOTHING();
+    }
+
+    void test_ref_layer(const std::vector<int>& a_shape, const std::vector<int>& b_shape, const String op)
```

Comment:
I think we should remove this function(or reuse the third one in both ref and non-ref with bool parameter isRef, that changes which layer we add), so that we avoid code duplication and different input generation between ref and non-ref functions.

Reply:
Done with reusing with a bool parameter `isRef`.